File: SenLib/Sen3/FileFixes/tk_zessica_dat.cs
using HyoutaUtils;
using HyoutaUtils.Streams;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SenLib.Sen3.FileFixes {
	class tk_zessica_dat : FileMod {
		public string GetDescription() {
			return "Fix typo in conversation with Jessica.";
		}

		public IEnumerable<FileModResult> TryApply(FileStorage storage) {
			var file = storage.TryGetDuplicate(new HyoutaUtils.Checksum.SHA1(0x8a9a5ca2e438e36eul, 0xa47c147124623407ul, 0xae9b6b5bu));
			if (file == null) {
				return null;
			}

			var bin = file.CopyToMemory();
			var patcher = new SenScriptPatcher(bin);

			// Nevermind -> Never mind
			patcher.ExtendPartialCommand(0x1bec, 0x9e, 0x1c83, new byte[] {0x20});

			return new FileModResult[] { new FileModResult("data/scripts/talk/dat_en/tk_zessica.dat", bin) };
		}

		public IEnumerable<FileModResult> TryRevert(FileStorage storage) {
			var file = storage.TryGetDuplicate(new HyoutaUtils.Checksum.SHA1(0x8a9a5ca2e438e36eul, 0xa47c147124623407ul, 0xae9b6b5bu));
			if (file == null) {
				return null;
			}
			return new FileModResult[] { new FileModResult("data/scripts/talk/dat_en/tk_zessica.dat", file) };
		}
	}
}

File: native/sen3/FileFixes/tk_zessica_dat.cpp
#include <filesystem>
#include <string_view>
#include <vector>

#include "file_getter.h"

#include "p3a/pack.h"
#include "p3a/structs.h"
#include "sen/sen_script_patcher.h"
#include "sha1.h"

namespace SenLib::Sen3::FileFixes::tk_zessica_dat {
std::string_view GetDescription() {
    return "Fix typo in conversation with Jessica.";
}

bool TryApply(const SenPatcher::GetCheckedFileCallback& getCheckedFile,
              std::vector<SenPatcher::P3APackFile>& result) {
    try {
        auto file = getCheckedFile(
            "data/scripts/talk/dat_en/tk_zessica.dat",
            15102,
            SenPatcher::SHA1FromHexString("8a9a5ca2e438e36ea47c147124623407ae9b6b5b"));
        if (!file) {
            return false;
        }

        auto& bin = file->Data;
        SenScriptPatcher patcher(bin);

        // Nevermind -> Never mind
        patcher.ExtendPartialCommand(0x1bec, 0x9e, 0x1c83, {{0x20}});

        result.emplace_back(std::move(bin), file->Filename, SenPatcher::P3ACompressionType::LZ4);

        return true;
    } catch (...) {
        return false;
    }
}
} // namespace SenLib::Sen3::FileFixes::tk_zessica_dat
